src/transformers/modeling_tf_utils.py

```diff
@@ -884,6 +885,93 @@ def load_repo_checkpoint(self, repo_path_or_name):
         # set it directly, but the user can pass it to fit().
         return {"epoch": extra_data["epoch"]}
 
+    def prepare_tf_dataset(
+        self,
+        dataset: Any,
```

Comment:
Since this function is to be used in tandem with `datasets`, perhaps this function could be entirely import-protected by `is_datasets_available`. Then we could put the right type here.

(Alternatively, you can use a [forward reference](https://peps.python.org/pep-0484/#forward-references) here. The information would be nice, but I don't know if our docs play well with that 🤔)

Reply:
Yes, please use a forward reference here.